DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js

```diff
@@ -46,42 +53,39 @@
                                 powerbi.data.SQExprBuilder.text(value));
                         return powerbi.data.createDataViewScopeIdentity(expr);
                     });
-
-            graphMetadata = {
-                columns: [
-                    {
-                        displayName: 'Time',
-                        isMeasure: true,
-                        queryName: 'timestamp',
-                        type: powerbi.ValueType.fromDescriptor({ dateTime: true })
-                    },
-                    {
-                        displayName: 'Humidity',
-                        isMeasure: true,
-                        format: "0.00",
-                        queryName: 'humidity',
-                        type: powerbi.ValueType.fromDescriptor({ numeric: true }),
-                    },
-                    {
-                        displayName: 'Temperature',
+                    
+            var graphMetadataColumns = [
+                {
+                    displayName: 'Time',
+                    isMeasure: true,
+                    queryName: 'timestamp',
+                    type: powerbi.ValueType.fromDescriptor({ dateTime: true })
+                }
+            ];
+            
+            columns = [];
+            
+            // Create a new column for non-reserved values
+            for (var field in data[0]) {
+                if (reservedDataColumns.indexOf(field) === -1) {
+                    graphMetadataColumns.push({
+                        displayName: field,
```

Comment:
The `displayName` will match the field name unless there is some way for the fields to specify their friendly name